src/App.tsx

```diff
@@ -1,61 +1,123 @@
+import React, { FormEventHandler, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [userId, setUserId] = useState<number>(0);
+  const [title, setTitle] = useState<string>('');
+  const [newTodos, setTodos] = useState<Todo[]>(todos);
+  const [userError, setUserError] = useState<string | null>(null);
+  const [titleError, setTitleError] = useState<string | null>(null);
+
+  const handleSubmit: FormEventHandler = (event) => {
+    event.preventDefault();
+
+    if (!userId && !title) {
+      setUserError('Please choose a user');
+      setTitleError('Please enter a title');
+
+      return;
+    }
+
+    // was title described?
+    if (!title) {
+      setTitleError('Please enter a title');
+
+      return;
+    }
+
+    // was user selected?
+    if (!userId) {
+      setUserError('Please choose a user');
+
+      return;
+    }
+
+    const userObj = usersFromServer.find(user => user.id === userId);
+
+    if (userObj) {
+      const newTodo: Todo = {
+        id: todos.length + 1,
```

Comment:
tests failed: 1) should add a title with the max existing id + 1 
so you need to find largest id and add 1 to it's value
[1, 4, 3, 2, 5, 32, 1] -> due to requirements id should be 33, but you would have 7 😉